ignite/pkg/cosmosgen/generate_typescript.go

```diff
@@ -121,90 +133,61 @@ func (g *tsGenerator) generateModuleTemplates() error {
 		}
 	}
 
-	add(g.g.appPath, g.g.appModules)
+	add(g.g.appPath, g.g.appModules, g.g.appIncludes)
 
 	// Always generate third party modules; This is required because not generating them might
 	// lead to issues with the module registration in the root template. The root template must
 	// always be generated with 3rd party modules which means that if a new 3rd party module
 	// is available and not generated it would lead to the registration of a new not generated
 	// 3rd party module.
 	for sourcePath, modules := range g.g.thirdModules {
-		add(sourcePath, modules)
+		add(sourcePath, modules.Modules, append(g.g.appIncludes, modules.Includes...))
 	}
 
 	return gg.Wait()
 }
 
 func (g *tsGenerator) generateModuleTemplate(
 	ctx context.Context,
+	protocCmd protoc.Cmd,
 	staCmd sta.Cmd,
-	appPath string,
+	tsprotoPluginPath, appPath string,
 	m module.Module,
+	includePaths []string,
 ) error {
 	var (
 		out      = g.g.o.jsOut(m)
 		typesOut = filepath.Join(out, "types")
-		conf     = swaggercombine.Config{
-			Swagger: "2.0",
-			Info: swaggercombine.Info{
-				Title: "HTTP API Console",
-			},
-		}
 	)
-
 	if err := os.MkdirAll(typesOut, 0o766); err != nil {
 		return err
 	}
 
 	// generate ts-proto types
-	if err := g.g.buf.Generate(
+	err := protoc.Generate(
 		ctx,
-		m.Pkg.Path,
 		typesOut,
-		g.tsTemplate(),
-		"module.proto",
-	); err != nil {
-		return err
-	}
-
-	// generate OpenAPI spec
-	tmp, err := os.MkdirTemp("", "gen-js-openapi-module-spec")
-	if err != nil {
-		return err
-	}
-
-	defer os.RemoveAll(tmp)
-
-	if err := g.g.buf.Generate(
-		ctx,
 		m.Pkg.Path,
-		tmp,
-		g.g.openAPITemplate(),
-		"module.proto",
-	); err != nil {
-		return err
-	}
-
-	// combine all swagger files
-	specs, err := xos.FindFiles(tmp, xos.JSONFile)
+		includePaths,
+		tsOut,
+		protoc.Plugin(tsprotoPluginPath, "--ts_proto_opt=snakeToCamel=true", "--ts_proto_opt=esModuleInterop=true"),
+		protoc.Env("NODE_OPTIONS="), // unset nodejs options to avoid unexpected issues with vercel "pkg"
+		protoc.WithCommand(protocCmd),
+	)
 	if err != nil {
 		return err
 	}
 
-	for _, spec := range specs {
-		if err := conf.AddSpec(strcase.ToCamel(m.Pkg.Name), spec); err != nil {
-			return err
-		}
-	}
+	specPath := filepath.Join(out, "api.swagger.yml")
+	fmt.Println(specPath)
```

Comment:
Forgotten print?

```suggestion
```

Reply:
I deleted that...just haven't pushed cause I was trying to fix the register routes issue